### PR TITLE
Re-produce the original 'useLiteralContent' setting

### LINKED ISSUE #6472: Metadata settings should re-produce the original 'useLiteralContents' setting from the compilation input.
Metadata settings should re-produce the original 'useLiteralContent' setting from the compilation input.

References https://github.com/ethereum/solidity/issues/6471

### PR DESCRIPTION
**Pull request by Marenz**
fixes #6472

### CODE REVIEW

File: libsolidity/interface/CompilerStack.cpp [hunk 1]
```diff
@@ -989,6 +989,7 @@ string CompilerStack::createMetadata(Contract const& _contract) const
 	meta["version"] = 1;
 	meta["language"] = "Solidity";
 	meta["compiler"]["version"] = VersionStringStrict;
+	meta["useLiteralContent"] = m_metadataLiteralSources;
```
Comment (axic) [inline libsolidity/interface/CompilerStack.cpp]: This should go under `settings`.

**Review by axic — CHANGES_REQUESTED** [review]

File: libsolidity/interface/CompilerStack.cpp (hunk 1 above)
Comment (Marenz) [reply libsolidity/interface/CompilerStack.cpp]: Just `settings`, or `metadata.settings`?

Comment (axic) [reply libsolidity/interface/CompilerStack.cpp]: Also needs to go into documentation.

Comment (Marenz) [reply libsolidity/interface/CompilerStack.cpp]: oh, nevermind, we _are_ in `metadata`

**Comment by Marenz** [comment]
Updated

File: libsolidity/interface/CompilerStack.cpp [hunk 2]
```diff
@@ -989,6 +989,7 @@ string CompilerStack::createMetadata(Contract const& _contract) const
 	meta["version"] = 1;
 	meta["language"] = "Solidity";
 	meta["compiler"]["version"] = VersionStringStrict;
+	meta["settings"]["useLiteralContent"] = m_metadataLiteralSources;
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.cpp]: Can you move it further down to where the other settings are populated?

**Comment by Marenz** [comment]
Moved it further down
Added it to documentation

File: test/libsolidity/Metadata.cpp [hunk 3]
```diff
@@ -177,6 +177,28 @@ BOOST_AUTO_TEST_CASE(metadata_relevant_sources_imports)
 	BOOST_CHECK(metadata["sources"].isMember("C"));
 }
 
+BOOST_AUTO_TEST_CASE(metadata_useLiteralContent)
+{
+	// Check that the metadata stamp is at the end of the runtime bytecode.
```
Comment (chriseth) [inline test/libsolidity/Metadata.cpp]: Is this comment still valid?

File: test/libsolidity/Metadata.cpp [hunk 4]
```diff
@@ -177,6 +177,28 @@ BOOST_AUTO_TEST_CASE(metadata_relevant_sources_imports)
...
+	compilerStack.setSources({{"", std::string(sourceCode)}});
+	compilerStack.setEVMVersion(dev::test::Options::get().evmVersion());
+	compilerStack.setOptimiserSettings(dev::test::Options::get().optimize);
+	BOOST_REQUIRE_MESSAGE(compilerStack.compile(), "Compiling contract failed");
+	string metadata_str = compilerStack.metadata("test");
+	Json::Value metadata;
+	jsonParse(metadata_str, metadata);
+	BOOST_CHECK(metadata.isMember("settings"));
+	BOOST_CHECK(metadata["settings"].isMember("useLiteralContent"));
+	BOOST_CHECK(!metadata["settings"]["useLiteralContent"].asBool());
```
Comment (chriseth) [inline test/libsolidity/Metadata.cpp]: Please also add another test that checks for true here.

**Comment by Marenz** [comment]
Fixed comment
Checking for false/true

**Comment by chriseth** [comment]
Please also add a changelog entry.

**Comment by Marenz** [comment]
Added changelog

**Review by chriseth — APPROVED** [review]

**Pull request merged**